shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -274,12 +282,19 @@ @interface FlutterEngineRegistrar : NSObject <FlutterPluginRegistrar>
 - (instancetype)initWithPlugin:(nonnull NSString*)pluginKey
                  flutterEngine:(nonnull FlutterEngine*)flutterEngine;
 
-- (NSView*)viewForId:(FlutterViewId)viewId;
+- (nullable NSView*)viewForId:(FlutterViewId)viewId;
+
+/**
+ * The value published by this plugin, or NSNull if nothing has been published.
+ *
+ * The unusual NSNull is for the documented behavior of valuePublishedByPlugin:.
+ */
+@property(nonatomic, readonly, nonnull) NSObject* publishedValue;
 @end
 
 @implementation FlutterEngineRegistrar {
   NSString* _pluginKey;
-  FlutterEngine* _flutterEngine;
+  __weak FlutterEngine* _flutterEngine;
```

Comment:
Was this intended to be in this patch? Either way, nice catch. If you wanted to test this, there's a similar test for the channels you could optionally clone and tweak.

Reply:
Ah I see the comment at the end of the description. Answer is yes :)